scripts/skills/perks/perk_rf_phalanx.nut

```diff
@@ -45,45 +51,40 @@ this.perk_rf_phalanx <- ::inherit("scripts/skills/skill", {
 
 	function onAnySkillUsed( _skill, _targetEntity, _properties )
 	{
-		if (_targetEntity != null && _skill.isAttack() && !_skill.isRanged())
+		if (_skill.isAttack() && !_skill.isRanged() && (this.m.RequiredDamageType == null || _skill.getDamageType().contains(this.m.RequiredDamageType)))
 		{
-			_properties.Reach += this.getCount();
+			_properties.Reach += this.getAllyCount();
 		}
 	}
 
 	function onBeingAttacked( _attacker, _skill, _properties )
 	{
 		if (_skill.isAttack() && !_skill.isRanged())
 		{
-			_properties.Reach += this.getCount();
+			_properties.Reach += this.getAllyCount();
 		}
 	}
 
-	function onAfterUpdate( _properties )
+	function onUpdate( _properties )
 	{
-		if (this.hasAdjacentShieldwall())
+		if (this.isActorValid(this.getContainer().getActor()))
 		{
-			local shieldwallSkill = this.getContainer().getSkillByID("actives.shieldwall");
-			if (shieldwallSkill != null)
-			{
-				shieldwallSkill.m.ActionPointCost = ::Math.max(shieldwallSkill.m.ActionPointCost * this.m.ShieldWallActionPointsMult, this.m.ShieldWallActionPointsMininum);
-				shieldwallSkill.m.FatigueCostMult *= this.m.ShieldWallFatigueMult;
-			}
+			_properties.StartSurroundCountAt += this.m.StartSurroundCountAtModifier * this.getAllyCount();
 		}
 	}
 
-	function getCount()
+	function getAllyCount()
 	{
 		local ret = 0;
 		local actor = this.getContainer().getActor();
-		if (!actor.isPlacedOnMap() || !actor.isArmedWithShield() || actor.getOffhandItem().getID().find("buckler") != null)
+		if (!actor.isPlacedOnMap() || !this.isActorValid(actor))
```

Comment:
I don't think we should check `!this.isActorValid(actor)` in here. With that check the `getAllyCount` function would do more than you would expect from it.
We might want to display the effect you would get, if you were to swap back to a valid weapon setup or become un-disarmed (if that was be to added as a condition)